documentation/src/main/asciidoc/reference/search-dsl-predicate.asciidoc

```diff
@@ -630,6 +630,71 @@ but can be <<search-dsl-predicate-common-boost,boosted>>,
 either on a per-field basis with a call to `.boost(...)` just after `.field(...)`/`.fields(...)`
 or for the whole predicate with a call to `.boost(...)` after `.matching(...)`.
 
+[[search-dsl-predicate-terms]]
+== `terms`: match a set of terms
+
+The `terms` predicate matches documents for which a given field contains some terms, any or all of them.
+
+With `matchingAny` we require that at least one of the provided terms matches.
+Functionally, this is somewhat similar to a <<search-dsl-predicate-boolean-or,boolean `OR`>> with one <<search-dsl-predicate-match,`match`>> predicate per term, but the syntax for a single `terms` predicate is more concise.
+
+NOTE: `matchingAny` expects to be passed *terms*, not just any string. The given terms will not be analyzed. See <<search-dsl-predicate-terms-analysis>>.
+
+.Matching any of the provided terms
+====
+[source, JAVA, indent=0, subs="+callouts"]
+----
+include::{sourcedir}/org/hibernate/search/documentation/search/predicate/PredicateDslIT.java[tags=terms-any]
+----
+====
+
+With `matchingAll` we require that all of the provided terms match.
+Functionally, this is somewhat similar to a <<search-dsl-predicate-boolean-and,boolean `AND`>> with one <<search-dsl-predicate-match,`match`>> predicate per term, but the syntax for a single `terms` predicate is more concise.
+
+NOTE: `matchingAll` expects to be passed *terms*, not just any string. The given terms will not be analyzed. See <<search-dsl-predicate-terms-analysis>>.
+
+[WARNING]
+====
+With `matchingAll` the given terms must not be more than 1024.
+====
+
+.Matching all of the provided terms
+====
+[source, JAVA, indent=0, subs="+callouts"]
+----
+include::{sourcedir}/org/hibernate/search/documentation/search/predicate/PredicateDslIT.java[tags=terms-any]
+----
+====
+
+[[search-dsl-predicate-terms-analysis]]
+=== Terms predicates and analysis
+
+NOTE: For an introduction to analysis and how to configure it,
+refer to the <<concepts-analysis>> section.
+
+Differently from other predicates, the terms passed to `matchingAny()` or `matchingAll()`
+are never analyzed nor **usually** normalized.
+If it is true, this means that in order to match the related documents,
+you need to provide the terms as they were written to the indexes.
+
+This is always true with the Lucene backend.
+For the Elasticsearch backend: terms are never analyzed nor normalized if an analyzer is defined on the field.
+While terms are normalized if a normalizer is defined on the field.
```

Comment:
A sentence should not start with "while".

Regardless, could you please try to be consistent with existing documentation? We already have a similar warning for the wildcard predicate, and we used an `IMPORTANT` admonition there. So here we should have something like this:

```suggestion
[IMPORTANT]
====
If an analyzer has been defined on the field, the terms will not be analyzed nor normalized.

If a normalizer has been defined on the field:

* when using the Elasticsearch backend, the terms **will** be normalized.
* when using the Lucene backend, the terms **will not** be normalized.

For example, the term `Cat` could match `cat`
when targeting a field having a normalizer that applies a lowercase filter when indexing,
but only when using the Elasticsearch backend.
When using the Lucene backend, only the term `cat` could match `cat`.
====
```